inference-engine/src/mkldnn_plugin/nodes/mkldnn_pad_node.cpp

```diff
@@ -0,0 +1,336 @@
+        int dstIdx = 0;
+        for (size_t i = 0; i < dimsSize_1; ++i)
+            dstIdx += counters[i] * params.dstStrides[i];
+
+        for (size_t iwork = start; iwork < end; ++iwork, dstIdx += inputSV) {
+            int srcIdx = 0;
+            for (size_t i = 0; i < dimsSize_1; ++i) {
+                int idx = (counters[i] < padsBegin[i]) ? 0 :
+                          ((counters[i] >= params.srcODims[i]) ? (srcDims[i] - 1) : (counters[i] - padsBegin[i]));
+                srcIdx += idx * params.srcStrides[i];
+            }
+
+            std::fill_n(&dstData[dstIdx], padsBegin[dimsSize_1], srcData[srcIdx]);
+            cpu_memcpy(&dstData[dstIdx + padsBegin[dimsSize_1]], &srcData[srcIdx],
+                       srcDims[dimsSize_1] * sizeof(float));
+            std::fill_n(&dstData[dstIdx + params.srcODims[dimsSize_1]], dstDims[dimsSize_1] - params.srcODims[dimsSize_1],
+                        srcData[srcIdx + srcDims[dimsSize_1] - 1]);
+
+            parallel_step(dimsSize_1, counters, dstDims);
+        }
+    });
+}
+
+void MKLDNNPadNode::padReflect(const float *srcData, float* dstData, SizeVector srcDims, SizeVector dstDims) {
+    SizeVector src_2;
+    for (size_t i = 0; i < srcDims.size(); i++)
+        src_2.push_back(srcDims[i] + params.srcODims[i] - 2);
+
+    size_t dimsSize_1 = dstDims.size() - 1;
+    size_t inputSV = dstDims[dimsSize_1];
+    size_t workAmountDst = params.dstStrides[0] * dstDims[0] / dstDims[dimsSize_1];
+
+    parallel_nt(0, [&](const int ithr, const int nthr) {
+        size_t start = 0, end = 0;
+        SizeVector counters(dimsSize_1, 0);
+        splitter(workAmountDst, nthr, ithr, start, end);
+
+        parallel_init(start, dimsSize_1, counters, dstDims);
+        int dstIdx = 0;
+        for (size_t i = 0; i < dimsSize_1; ++i)
+            dstIdx += counters[i] * params.dstStrides[i];
+
+        for (size_t iwork = start; iwork < end; ++iwork, dstIdx += inputSV) {
+            int srcIdx = 0;
+            for (size_t i = 0; i < dimsSize_1; ++i) {
+                int idx = (counters[i] < padsBegin[i]) ? (padsBegin[i] - counters[i]) :
+                          ((counters[i] >= params.srcODims[i]) ? (src_2[i] - counters[i]) : (counters[i] - padsBegin[i]));
+                srcIdx += idx * params.srcStrides[i];
+            }
+
+            for (size_t i = 0; i < padsBegin[dimsSize_1]; ++i)
+                dstData[dstIdx + i] = srcData[srcIdx + padsBegin[dimsSize_1] - i];
+
+            cpu_memcpy(&dstData[dstIdx + padsBegin[dimsSize_1]], &srcData[srcIdx], sizeof(float) * srcDims[dimsSize_1]);
+
+            for (size_t i = params.srcODims[dimsSize_1]; i < dstDims[dimsSize_1]; ++i)
+                dstData[dstIdx + i] = srcData[srcIdx + src_2[dimsSize_1] - i];
+
+            parallel_step(dimsSize_1, counters, dstDims);
+        }
+    });
+}
+
+void MKLDNNPadNode::padSymmetric(const float *srcData, float* dstData, SizeVector srcDims, SizeVector dstDims) {
```

Comment:
It looks like we can unite `padReflect()` and `padSymmetric()`

Reply:
Agree. I would also vote for unification.